### PR TITLE
Use portable-atomic in frb_rust

### PR DESCRIPTION
**Pull request by kyrias**
Not all targets supports `AtomicU64` which currently makes it difficult to share the same crate between multiple targets. `portable-atomic` uses the std version of the type if one exists and otherwise emulates a fallback using atomic CAS operations.

## Changes

_Please list issues fixed by this PR here, using format "Fixes #the-issue-number"._

## Checklist

- [ ] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to add a test.
- [ ] `./frb_internal precommit --mode slow` (or `fast`) is run (it internal runs code generator, does auto formatting, etc).
- [ ] If this PR adds/changes features, documentations (in the `./website` folder) are updated.
- [ ] CI is passing. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to solve a failed CI.

## Remark for PR creator

- `./frb_internal --help` shows utilities for development.
- If fzyzcjy does not reply for a few days, maybe he just did not see it, so please ping him.

### CODE REVIEW

File: frb_rust/Cargo.toml
[the remark this answers is not in the document]
Comment (kyrias) [reply frb_rust/Cargo.toml:1]: Hmm.  It's a pretty minimal dependency, but that's fair.  Maybe optional but enabled by default?

**Comment by kyrias** [comment]
I'm so confused here.  I push something and some CI jobs fail.  I run those locally and commit and push the `pubspec.lock` changes it complained about which then causes _another_ CI job to fail because it wants the `pubspec.lock` file to be unchanged?

File: frb_rust/Cargo.toml
[the remark this answers is not in the document]
Comment (fzyzcjy) [reply frb_rust/Cargo.toml:1]: Sure, we should enable it by default.

**Comment by fzyzcjy** [comment]
@all-contributors please add @kyrias for code

**Pull request merged**